### PR TITLE
vmadm: improvements

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Some refactoring/pythonification.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/modules/cloud/smartos/vmadm.py

### CODE REVIEW

File: plugins/modules/cloud/smartos/vmadm.py [hunk 1]
```diff
@@ -541,20 +542,15 @@ def set_vm_state(module, vm_uuid, vm_state):
...
+    cmd = [module.vmadm, command] + force + [vm_uuid]
 
     (rc, stdout, stderr) = module.run_command(cmd)
 
     match = re.match('^Successfully.*', stderr)
-    if match:
-        return True
-    else:
-        return False
+    return bool(match)
```
Comment (felixfontein) [inline plugins/modules/cloud/smartos/vmadm.py:553]: How about:
```suggestion
    return match is not None
```

File: plugins/modules/cloud/smartos/vmadm.py [hunk 2]
```diff
@@ -601,23 +597,17 @@ def vm_state_transition(module, uuid, vm_state):
 
 
 def is_valid_uuid(uuid):
-    if re.match('^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$', uuid, re.IGNORECASE):
-        return True
-    else:
-        return False
+    return bool(re.match('^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$', uuid, re.IGNORECASE))
```
Comment (felixfontein) [inline plugins/modules/cloud/smartos/vmadm.py:600]: Here's a very similar one:
```suggestion
    return re.match('^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$', uuid, re.IGNORECASE) is not None
```

File: plugins/modules/cloud/smartos/vmadm.py [hunk 3]
```diff
@@ -601,23 +597,17 @@ def vm_state_transition(module, uuid, vm_state):
...
-    failed = []
-
-    for u in [['uuid', module.params['uuid']],
-              ['image_uuid', module.params['image_uuid']]]:
-        if u[1] and u[1] != '*':
-            if not is_valid_uuid(u[1]):
-                failed.append(u[0])
+    failed = [
+        name
+        for name, pvalue in [['uuid', module.params['uuid']], ['image_uuid', module.params['image_uuid']]]
```
Comment (russoz) [inline plugins/modules/cloud/smartos/vmadm.py:606]:
```suggestion
        for name, pvalue in [(x, module.params[x]) for x in ['uuid', 'image_uuid']]
```

**Review by felixfontein — APPROVED** [review]
Looks good to me. Let's see what the module maintainers say.

**Comment by felixfontein** [comment]
@russoz thanks for working on this!

**Pull request merged**